File: app/src/main/java/com/themobilecompany/memorygame/persistence/ScoreDao.java
package com.themobilecompany.memorygame.persistence;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.themobilecompany.memorygame.model.Score;

import java.util.List;

import io.reactivex.Flowable;

@Dao
public interface ScoreDao {

    @Query("SELECT * FROM scores ORDER BY scorePoints DESC")
    Flowable<List<Score>> getAllScores();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertScore(Score score);

    @Query("DELETE FROM scores")
    void deleteAllScores();
}

File: app/src/main/java/com/games/memorygame/persistence/ScoreDao.kt
package com.games.memorygame.persistence

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.OnConflictStrategy
import android.arch.persistence.room.Query

import com.games.memorygame.model.Score

import io.reactivex.Flowable

@Dao
interface ScoreDao {

    @get:Query("SELECT * FROM scores ORDER BY scorePoints DESC")
    val allScores: Flowable<List<Score>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertScore(score: Score)

    @Query("DELETE FROM scores")
    fun deleteAllScores()
}
